diagrams/library-gen.puml

```diff
@@ -0,0 +1,46 @@
+@startuml
```

Comment:
This file is generated, so I assume it does not need to be reviewed

Reply:
(In a couple of PRs, I'll add generation of Mermaid diagrams, so you could see these diagrams directly on GitHub.)